### PR TITLE
Added support to return device registration details in all the cases

### PR DESCRIPTION
**Pull request by valangot**
## Proposed changes

Added support to return device registration details in all the cases.
We have seen getDeviceInfo API gets failed sometime due to SSO extension

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: MSAL/src/instance/MSALDeviceInfoProvider.m
Comment (oldalton) [inline MSAL/src/instance/MSALDeviceInfoProvider.m:46]: Should we check for error first and return it prior to creating default info?

File: MSAL/src/instance/MSALDeviceInfoProvider.m
Comment (oldalton) [inline MSAL/src/instance/MSALDeviceInfoProvider.m:64]: If you're not using this error info, no need to create it. You can just switch it to logging.

File: MSAL/src/instance/MSALDeviceInfoProvider.m
Comment (oldalton) [inline MSAL/src/instance/MSALDeviceInfoProvider.m:58]: You shouldn't return both object and the error.

**Review by oldalton — CHANGES_REQUESTED** [review]

File: MSAL/src/instance/MSALDeviceInfoProvider.m
Comment (valangot) [reply MSAL/src/instance/MSALDeviceInfoProvider.m:64]: Yes we are not using ErrorInfo and we can remove and rely on logs but it will break other clients if they are using it.

Mostly we follows the below coding practice

if (error) {
 / /Handle Error
} else {
 // Process result
}

from Defender, We don't check error and instead directly check into the Dictionary for registration details but other client must be using above practice as they are interested in DeviceMode and ssoStatus

File: MSAL/src/instance/MSALDeviceInfoProvider.m
Comment (valangot) [reply MSAL/src/instance/MSALDeviceInfoProvider.m:58]: Option -1: Will be very straight forward and will returns registration state without disturbing deviceState and SSO extension.

Option-2: Mostly we have seen failures from line#98 i.e. MSIDSSOExtensionGetDeviceInfoRequest::executeRequestWithCompletion returns error so if we go with return msalDeviceInfo = nil, error. then its as if we are not adding any value and API will failed to return registration state due to SSO extension failures.

Or can we add some enum or scope so that caller will explicitly mention for WPJ state?

Comment (oldalton) [reply MSAL/src/instance/MSALDeviceInfoProvider.m:58]: @valangot, @jasoncoolmax, could you discuss and conclude?

Comment (valangot) [reply MSAL/src/instance/MSALDeviceInfoProvider.m:57]: Discussed offline with @jasoncoolmax, returning devcieInfo and error from couple of places, We will monitor errors and AppCenter Defender telemetry and try to resolve SSO requests intermittent issue.

cc @oldalton

Comment (oldalton) [reply MSAL/src/instance/MSALDeviceInfoProvider.m:57]: @valangot, I've dismissed my review, so feel free to merge once you get an approval from @jasoncoolmax

File: MSAL/src/instance/MSALDeviceInfoProvider.m
Comment (jasoncoolmax) [inline MSAL/src/instance/MSALDeviceInfoProvider.m:90]: ssoExtensionRequest being nil is an internal critical error, we may want to keep it as it is :)
 `completionBlock(nil, requestError);`

File: MSAL/src/instance/MSALDeviceInfoProvider.m
Comment (jasoncoolmax) [inline MSAL/src/instance/MSALDeviceInfoProvider.m:44]: will it be better to call it `fillDeviceInfoCompletionBlock`? Besides, we use `handler` to mean something else in our code :)

**Review by jasoncoolmax — APPROVED** [review]

**Comment by jasoncoolmax** [comment]
Pipeline `[Broker] MSAL submodule update check` failed maybe because it is merging from a fork.

I verified the broker build manually. So I force merge the PR here.

**Pull request merged**
